snapcraft/cli/__init__.py

```diff
@@ -89,6 +89,7 @@ def get_command(self, ctx, cmd_name):
 @click.option('--debug', '-d', is_flag=True)
 def run(ctx, debug, catch_exceptions=False, **kwargs):
     """Snapcraft is a delightful packaging tool."""
+    ctx.obj['debug'] = debug
     if debug:
```

Comment:
Ok, you go what you wanted, I was mistaken on the behavior of `get_current_context`.

Please validate it all still works.